src/go/rpk/pkg/cli/cmd/container/start.go

```diff
@@ -221,18 +227,18 @@ func startCluster(c common.Client, n uint) error {
 			mu.Lock()
 			nodes = append(nodes, node{
 				id,
-				fmt.Sprintf(
-					"%s:%d",
-					state.ContainerIP,
-					state.HostKafkaPort,
-				),
+				nodeAddr(state.HostKafkaPort),
```

Comment:
127.0.0.1 is what's used from outside the container. The actual container IP is used in the redpanda config in each node, since using 127.0.0.1 would be interpreted as that container's localhost, resulting in weird errors.

Reply:
Sounds reasonable.
nit: We could make it more apparent by renaming the method to sth like `externalNodeAddr` :thinking: